pyswmm/simulation.py

```diff
@@ -818,3 +828,239 @@ def test_callback():
+    .. code-block:: python
+
+        [SUBCATCHMENTS]
+        ;;                                         Total   Pcnt. Pcnt.  Curb
+        ;;Name   Raingage         Outlet   Area    Imperv  Width Slope  Length
+        ;;------ ---------------- -------- ------- ------- ----- ------ --------
+        S1       SCS_24h_Type_I_1in J1     1       100     500   0.5    0
+
+
+    .. code-block:: python
+
+        from pyswmm import Simulation, SimulationPreConfig, Subcatchments
+
+        # Create Config Handle
+        sim_conf = SimulationPreConfig()
+
+        # Specifying the update parameters
+        # Parameter Order:
+        # New Value, Section, Object ID, Parameter Index, Obj Row Num (optional)
+        sim_conf.add_update_by_token("J2", "SUBCATCHMENTS", "S1", 2)
+        sim_conf.add_update_by_token(2, "TIMESERIES", "SCS_24h_Type_I_1in", 2, 5)
+
+        with Simulation(<path-to-inp>, sim_preconfig = sim_conf) as sim:
+            S1 = Subcatchments(sim)["S1"]
+            print(S1.connection)
+
+            for step in sim:
+                pass
+
+    .. code-block:: python
+
+        >>> (2, 'J2')
+
+    In the New INP file:
+
+    .. code-block:: python
+
+        [SUBCATCHMENTS]
+        ;;                                         Total   Pcnt. Pcnt.  Curb
+        ;;Name   Raingage         Outlet   Area    Imperv  Width Slope  Length
+        ;;------ ---------------- -------- ------- ------- ----- ------ --------
+        S1       SCS_24h_Type_I_1in J3     1       100     500   0.5    0
+
+    """
+
+    def __init__(self):
+        self._filename_suffix = "_mod"
+        self._modifications = {}
+        self._source_input_name = None
+
+    @property
+    def input_file(self):
+        """
+        This is set by the `Simulation` class but can also be set directly
+        if the user wants to simply use this class for find replace.
+
+        Examples:
+
+        .. code-block:: python
+
+            sim_conf = SimulationPreConfig()
+            sim_conf.input_file = "./model_weir_setting.inp"
+
+        .. code-block::
+
+            >>> datetime.datetime(2016,5,10,15,15,1)
+        """
+        self._source_input_name
+
+    @input_file.setter
+    def input_file(self, inp_path):
+        """"""
+        self._source_input_name = inp_path
+
+    @property
+    def filename_suffix(self):
+        """
+        If the user wants to modify the new file name simply use this class
+        for find replace.
+
+        Examples:
+
+        .. code-block:: python
+
+            sim_conf = SimulationPreConfig()
+            sim_conf.filename_suffix = "_a"
+
+        .. code-block::
+
+            >>> datetime.datetime(2016,5,10,15,15,1)
+        """
+        return self._filename_suffix
+
+    @filename_suffix.setter
+    def filename_suffix(self, suffix: str):
+        """"""
+        self._filename_suffix = suffix
+
+    def add_update_by_token(self, new_val, section: str, id: str,
+                            index: int, row_num=0):
```

Comment:
Stylistically I take issue with the order of the arguments here. I think hierarchically, so it'd read better to me as written below. Also, putting all arguments on seperate lines makes future git diffs cleaner and less likely to have conflicts.

`
def add_update_by_token(
    self,
    section: str,
    id: str,
    index: int, 
    new_val, 
    row_num=0
):
`